gdal/swig/python/setup.py

```diff
@@ -463,7 +463,8 @@ def parallel_build_extensions(self):
     data_files=data_files,
     ext_modules=ext_modules,
     scripts=glob('scripts/*.py'),
-    cmdclass={'build_ext': gdal_ext}
+    cmdclass={'build_ext': gdal_ext},
+    extras_require={'numpy': ['numpy > 1.0.0']},
```

Comment:
shouldn't that be conditionalized by if HAVE_NUMPY: ?

Reply:
No, it's the opposite: using `pip install gdal[numpy]` will install `numpy`, so that `HAVE_NUMPY` will be true when building/installing `gdal`.